src/main/java/org/apache/commons/imaging/formats/tiff/TiffRasterData.java

```diff
@@ -117,42 +128,37 @@ public TiffRasterStatistics getSimpleStatistics() {
      * @param valueToExclude exclude samples with this specified value.
      * @return a valid instance.
      */
-    public TiffRasterStatistics getSimpleStatistics(final float valueToExclude) {
-        return new TiffRasterStatistics(this, valueToExclude);
-    }
+    public abstract TiffRasterStatistics getSimpleStatistics(float valueToExclude);
 
     /**
-     * Gets the width (number of columns) of the raster.
+     * Returns the content stored as an array in this instance. Note that in
+     * many cases, the returned array is <strong>not</strong> a safe copy of the
+     * data but a direct reference to the member element. In such cases,
+     * modifying it would directly affect the content of the instance. While
+     * this design approach carries some risk in terms of data security, it was
+     * chosen for reasons of performance and memory conservation. TIFF images
+     * that contain floating-point data are often quite large. Sizes of 100
+     * million raster cells are common. Making a redundant copy of such a large
+     * in-memory object might exceed the resources available to a Java
+     * application.
      *
-     * @return the width of the raster
+     * @return the data content stored in this instance.
      */
-    public int getWidth() {
-        return width;
-    }
+    public abstract float[] getData();
 
     /**
-     * Gets the height (number of rows) of the raster.
+     * Returns the content stored as an array in this instance. Note that in
+     * many cases, the returned array is <strong>not</strong> a safe copy of the
+     * data but a direct reference to the member element. In such cases,
+     * modifying it would directly affect the content of the instance. While
+     * this design approach carries some risk in terms of data security, it was
+     * chosen for reasons of performance and memory conservation. TIFF images
+     * that contain floating-point data are often quite large. Sizes of 100
+     * million raster cells are common. Making a redundant copy of such a large
+     * in-memory object might exceed the resources available to a Java
+     * application.
```

Comment:
Oh, the same comment here. @gwlucastrig what if we leave this comment only for the top-level class Javadoc comment, and either mention it briefly in these methods (maybe ask the reader to refer to the top-section) or just leave the top one and don't mention/duplicate the information here?

Reply:
done